symengine/integer.h

```diff
@@ -27,18 +27,15 @@ class Integer : public Number
     // explicit Integer(integer_class i);
     Integer(const integer_class &_i)
         : i(_i){SYMENGINE_ASSIGN_TYPEID()} Integer(integer_class && _i)
-        : i(std::move(_i))
-    {
-        SYMENGINE_ASSIGN_TYPEID()
-    }
-    //! \return size of the hash
-    virtual hash_t __hash__() const;
+        : i(std::move(_i)){SYMENGINE_ASSIGN_TYPEID()}
+          //! \return size of the hash
+          hash_t __hash__() const override;
```

Comment:
Hmm, not sure what's going on here.

Reply:
Looks like the macro is confusing clang-format, maybe it doesn't expand it? If I add a `;` and re-run clang-format it looks more sensible:

```
    Integer(const integer_class &_i) : i(_i)
    {
        SYMENGINE_ASSIGN_TYPEID();
    }
    Integer(integer_class &&_i) : i(std::move(_i))
    {
        SYMENGINE_ASSIGN_TYPEID();
    }
    //! \return size of the hash
    hash_t __hash__() const override;
```

Reply:
..but going from the existing convention for macros
```
#define SYMENGINE_MACRO() int x; 
// example of use:
SYMENGINE_MACRO()
```
to this
```
#define SYMENGINE_MACRO() int x
// example of use:
SYMENGINE_MACRO();
```
everywhere to improve the clang-fomat output would be a lot of changes - not sure that is wanted? but if so should probably be a separate PR?